openedx/adg/lms/webinars/models.py

```diff
@@ -302,6 +303,42 @@ def publish(self):
             self.save()
 
         self._send_webinar_update_or_invite_emails()
+        self._update_webinar_team_registrations_and_schedule_reminder_emails()
+
+    def _update_webinar_team_registrations_and_schedule_reminder_emails(self):
+        """
+        Update the webinar team registrations.
+
+        Create WebinarRegistration objects with `is_team_member_registration` field set
+        for newly added panelists and alternative hosts. Schedule email reminders for the
+        newly added team members.
+
+        Unset the `is_team_member_registration` field for removed panelists and alternative hosts.
+        Cancel any scheduled email reminders for the removed members.
+        """
+        old_values = getattr(self, '_loaded_values', {})
+
+        webinar_team_fields = ['panelists', 'alternative_hosts']
+        has_webinar_team_changed = has_any_field_changed(self, old_values, webinar_team_fields)
+
+        if not has_webinar_team_changed:
+            return
+
+        old_team_emails = set(
+            extract_emails_from_string(old_values.get('panelists')) +
+            extract_emails_from_string(old_values.get('alternative_hosts'))
+        )
```

Comment:
What will happen in case a webinar is being published for the first time, This will not create registrations as the fields are not changed. Pls add a check that in case a webinar was not published then create registration and if it was published then check for the change in fields.

Reply:
It will work.

Reply:
When it is published for the first time, the old values will be None whereas the current values will be different so it will work.

Reply:
Pls test this way.

Create a webinar with some panelists and invitees and then save it.
Publish it from the list view.
Registrations will not be created.

Reply:
I understand the scenario now. Good catch! 👍🏼